jerry-ext/include/jerryscript-ext/module.h

```diff
@@ -59,31 +59,109 @@ typedef struct jerryx_native_module_t
 #define JERRYX_NATIVE_MODULE(module_name, on_resolve_cb)  \
   JERRYX_NATIVE_MODULE_IMPLEM(module_name, on_resolve_cb)
 
-#define JERRYX_NATIVE_MODULE_IMPLEM(module_name, on_resolve_cb)        \
-  static jerryx_native_module_t _ ## module_name ## _definition =      \
-  {                                                                    \
-    .name_p = (jerry_char_t *) #module_name,                           \
-    .on_resolve_p = (on_resolve_cb),                                   \
-    .next_p = NULL                                                     \
-  };                                                                   \
-                                                                       \
+/**
+ * @brief  Defines the symbol name. This is used in any place where the generated symbol name
+ *         might be needed.
+ * 
+ * @param module_name: Name of the module.
+ * @returns: A concatenated identifier, representing the symbol used for the native module's structure.
+ */
+#define JERRYX_NATIVE_MODULE_SYMBOL(module_name) \
+  _ ## module_name ## _definition
+
+/**
+ * @brief  Generates a structure of `jerryx_native_module_t`.
+ * 
+ * @param module_name: Name of the module.
+ * @param on_resolve_cb: The function to be called once the module is resolved.
+ * @returns: A generated structure, representing a native module.
+ * 
+ * @see: jerryx_native_module_register, jerryx_native_module_unregister
+ * 
+ * @note: `const` is used to indicate that this structure shall not change during runtime.
+ */
+#define JERRYX_NATIVE_MODULE_STRUCTURE(module_name, on_resolve_cb)    \
+  const jerryx_native_module_t _ ## module_name ## _definition =      \
+  {                                                                   \
+    .name_p = (jerry_char_t *) #module_name,                          \
+    .on_resolve_p = (on_resolve_cb),                                  \
+    .next_p = NULL                                                    \
+  };
+
+/**
+ * @brief  Generates the definition AND function body for the initial constructor.
+ * 
+ * @param module_name: Name of the module.
+ * @returns: The generated, initial constructor.
+ * 
+ * @note: When compiled with ENABLE_INIT_FINI, you may - dependent on platform - enable
+ *        the initial constructuion when this module has been loaded as a shared object
+ *        or library. However, this does **not** apply to statically "compiled in" modules.
```

Comment:
Compiled-in modules can also have constructors. They will be executed before `main()`.

Reply:
Look at the tests for modules. Each test consists of a single executable, in which the modules auto-register on startup by way of the constructors. It even handles the case where constructors are disabled: https://github.com/pando-project/jerryscript/blob/master/tests/unit-ext/module/jerry-module-test.c#L192

Reply:
Oh! I honestly wasn't exactly ware of how that macro worked, so I looked up the attributes and used what I found to describe it in the doc. But - good to know! I will fix that.